### ISSUE TITLE
vSphere CSI Driver does not support Kubernetes 1.22

### ISSUE THREAD

**Issue by PennyScissors** [rancher/rancher#33848]
The vSphere CSI Driver does not support Kubernetes 1.22 due to unsupported v1beta1 CRD APIs. Support will be added at a later release ([support matrix](https://vsphere-csi-driver.sigs.k8s.io/compatiblity_matrix.html#compatibility-matrix-for-vsphere-csi-driver)), but in the meantime users with the `CSIMigrationvSphere` feature enabled should be warned not to upgrade to Kubernetes 1.22 in our release notes.

Kubernetes v1.22 changelog: [CSI known issues](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.22.md#csimigrationvsphere-feature-gate-has-not-migrated-to-new-crd-apis)

Blocked by:
*  https://github.com/kubernetes/cloud-provider-vsphere/issues/504

**Comment by rayandas** [comment]
@PennyScissors We can update vSphere CSI chart to `v2.4.0` as it supports K8s 1.22. This should fix this issue.

**Comment by PennyScissors** [comment]
@rayandas that works, but check if there is a new CPI that is also compatible with 1.22 to pair it with it since our existing does not support it.

**Comment by rayandas** [comment]
@PennyScissors yes there's a new CPI compatible with 1.22. Will update both CPI and CSI charts.

**Comment by doflamingo721** [comment]
### Update vSphere CPI and CSI to support kubernetes 1.22.
#### Test instructions:
- Rancher server: v2.6-head
- Deploy a downstream cluster using the vSphere Cloud Provider with k8s version 1.22

Fresh install checks:
- Install CPI `100.2.0+up1.1.0` and CSI `100.2.0+up2.4.1-rancher1` from dev repository.
- Perform fresh install checks (post the installation).

Upgrade cheks:
- Install CPI `100.1.0+up1.0.100` and CSI `CSI 100.1.0+up2.3.0` and upgrade CPI to `100.2.0+up1.1.0` and CSI to `100.2.0+up2.4.1-rancher1` and make sure all pods are up and active.

### Regression: Tolerations in vSphere CPI are incorrect
#### Test instructions:
- Rancher server: v2.6-head
- Deploy downstream RKE2 cluster using the vSphere Cloud Provider with k8s version 1.22
- Install CPI chart and check all the pods are active/running.
- Install CSI chart and check the if the CSI controller is deployed successfully in Deployments.
- Check if the following tolerations are present for both CPI daemonset and CSI controller deployment: 
````
        - key: node-role.kubernetes.io/control-plane
          effect: NoSchedule
          operator: Exists
        - key: node-role.kubernetes.io/etcd
          operator: Exists
          effect: NoExecute
````
### Other
#### For RKE2, test the following use cases:
1. When provisioning a cluster, use the cloud provider drop down and select "vsphere". These should use the charts packaged with RKE2 through KDM.
2. When provisioning a cluster, use the cloud provider drop down and select "external", then deploying the charts through apps & marketplace.

cc: @vivek-infracloud @rishabhmsra

**Comment by sirredbeard** [comment]
Dev validate and release note.

**Comment by slickwarren** [comment]
I was able to test this using 2.6.4-rc13:
* deploy downstream cluster (1cp, 1etcd, 2worker nodes) on 1.22.7-rancher1-1
* deploy cpi 100.2.0+up1.1.0 -- pass
* deploy csi 100.2.0+up2.4.1-rancher1 -- pass
* deploy a workload and mount a pv to it -- pass
* write data to the volume/check in datastore (through vsphere) -- pass

**Comment by braunsonm** [comment]
The release notes for 2.6.4 still state that vSphere does not support Kubernetes 1.22, but this was closed ahead of the release. Does this still block Kubernetes 1.22 in 2.6.4?

**Comment by phillipsj** [comment]
This was merged in and we have 1.22 support.
